### ISSUE TITLE
Add `edge_node` to `resource_arm_hdinsight_hadoop_cluster` at cluster creation time.

### ISSUE THREAD

**Issue by timehaven** [terraform-providers/terraform-provider-azurerm#4035]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

We would like to add an `edge_node` with a non-admin user to a Hadoop cluster at creation time so that a worker node exists that can submit Hadoop jobs. 

### New or Affected Resource(s)

* `resource_arm_hdinsight_hadoop_cluster`

### Potential Terraform Configuration

Adding a new `edge_node` block below the existing `zookeeper_node` block shown in the example at [hdinsight_hadoop_cluster](https://www.terraform.io/docs/providers/azurerm/r/hdinsight_hadoop_cluster.html).  The proposed new block is exactly like that in [hdinsight_rserver_cluster](https://www.terraform.io/docs/providers/azurerm/r/hdinsight_rserver_cluster.html) and the code would be similarly easy.  (Note the `zookeeper_node` block below already exists.)

```hcl
...
    zookeeper_node {
      vm_size  = "Standard_D3_V2"
      username = "acctestusrvm"
      password = "AccTestvdSC4daf986!"
    }

    edge_node {
      vm_size  = "Standard_D3_v2"
      username = "acctestusrvm"
      password = "AccTestvdSC4daf986!"
    }
...
```

### References

* For a start, existing code in the `resource_arm_hdinsight_rserver_cluster.go` [here](https://github.com/terraform-providers/terraform-provider-azurerm/blob/c331adb611fa5e1f8658be7e878ab4f7eb95aeac/azurerm/resource_arm_hdinsight_rserver_cluster.go#L42) and [here](https://github.com/terraform-providers/terraform-provider-azurerm/blob/c331adb611fa5e1f8658be7e878ab4f7eb95aeac/azurerm/resource_arm_hdinsight_rserver_cluster.go#L93) could be modified and added to `resource_arm_hdinsight_hadoop_cluster.go` below [here](https://github.com/terraform-providers/terraform-provider-azurerm/blob/c331adb611fa5e1f8658be7e878ab4f7eb95aeac/azurerm/resource_arm_hdinsight_hadoop_cluster.go#L41) and [here](https://github.com/terraform-providers/terraform-provider-azurerm/blob/c331adb611fa5e1f8658be7e878ab4f7eb95aeac/azurerm/resource_arm_hdinsight_hadoop_cluster.go#L91).

**Comment by rk-tbrigley** [comment]
@tombuildsstuff edgenode was added in https://github.com/terraform-providers/terraform-provider-azurerm/pull/4550 so this could be a dupe but!! could we see this edgenode functionality added to the other azurerm_hdinsight_* resources not just azurerm_hdinsight_hadoop_cluster?

**Comment by m4r74** [comment]
Are there any updates on edge nodes for other HDInsight clusters, i.e. Spark? Are there any blockers?

**Comment by tombuildsstuff** [comment]
👋🏻

Taking a look through here as @rk-tbrigley has mentioned this has been fixed via #4550 - and as such I'm going to close this issue for the moment.

Since this issue's been resolved, rather than extending the scope of this issue - would you mind opening a new issue to track adding the Edge Node to the specific HDInsight resources you're looking for?

Thanks!
